components/cam/src/dynamics/se/inidat.F90

```diff
@@ -453,31 +453,32 @@ subroutine read_inidat( ncid_ini, ncid_topo, dyn_in)
 
     if ( (ideal_phys .or. aqua_planet)) then
        tmp(:,1,:) = 0._r8
-       phis_tmp(:,:) = 0._r8
+       if (fv_nphys > 0) phis_tmp(:,:) = 0._r8
     else    
       fieldname = 'PHIS'
       tmp(:,1,:) = 0.0_r8
       if (fv_nphys > 0) then
-         ! Load phis field to GLL grid first
+         ! Load phis field to physics grid
          call infld(fieldname, ncid_topo, 'ncol', 1, nphys_sq, &
-              1, nelemd, phis_tmp, found, gridname='physgrid_d')
+                    1, nelemd, phis_tmp, found, gridname='physgrid_d')
       else
-         call infld(fieldname, ncid_topo, ncol_name,      &
-            1, npsq, 1, nelemd, tmp(:,1,:), found, gridname=grid_name)
-      endif
+         call infld(fieldname, ncid_topo, ncol_name, 1, npsq, &
+                    1, nelemd, tmp(:,1,:), found, gridname=grid_name)
+      end if ! fv_nphys > 0
       if(.not. found) then
          call endrun('Could not find PHIS field on input datafile')
       end if
     end if
 
-    if (fv_nphys == 0) then
-      ! Map phis data to FV physics grid
+    if (fv_nphys > 0) then
```

Comment:
File change shows me that the topography mapping used to be done from phys to dyn when grid_type = GLL. Now the same operation is done for pg2. Are you correcting a typo here?

Reply:
yes, that's correct.